gempy/library/nddops.py

```diff
@@ -44,35 +44,6 @@
 ONE = DQ.datatype(DQ.bad_pixel)
 
 
-def take_along_axis(arr, ind, axis):
-    """
-    Returns a view of an array (arr), re-ordered along an axis according to
-    the indices (ind). Taken from numpy issue 8708 under BSD licence,
-    pending inclusion into numpy (added in version 1.15.0).
-    """
-    if arr is None:
-        return None
-    if axis < 0:
-        if axis >= -arr.ndim:
-            axis += arr.ndim
-        else:
-            raise IndexError('axis out of range')
-    ind_shape = (1,) * ind.ndim
-    ins_ndim = ind.ndim - (arr.ndim - 1)   #inserted dimensions
-
-    dest_dims = list(range(axis)) + [None] + list(range(axis+ins_ndim, ind.ndim))
-
-    # could also call np.ix_ here with some dummy arguments, then throw those results away
-    inds = []
-    for dim, n in zip(dest_dims, arr.shape):
-        if dim is None:
-            inds.append(ind)
-        else:
-            ind_shape_dim = ind_shape[:dim] + (-1,) + ind_shape[dim+1:]
-            inds.append(np.arange(n).reshape(ind_shape_dim))
-    return arr[tuple(inds)]
-
-
```

Comment:
Removed this, since I also need `np.put_along_axis` and Astropy 4.0 requires Numpy 1.16 anyway. So I replaced this function below with `np.take_along_axis`, but interestingly the Numpy version does not expand dimensions when needed, so I needed to handle this with `np.expand_dims`.